src/components/dev-dot-content/mdx-components/mdx-headings/index.tsx

```diff
@@ -48,14 +50,24 @@ export function makeMdxHeadingElement(level: HeadingProps['level']) {
 	const { size, weight } = HEADING_LEVELS_TO_PROPS[level]
 
 	return function MdxHeading(props) {
+		const { children, ...rest } = props
 		return (
 			<Heading
-				{...props}
+				{...rest}
 				level={level}
 				className={fixedClassName}
 				size={size}
 				weight={weight}
-			/>
+			>
+				{/* Temporary permalink render until we update our remark plugin */}
+				{Children.map(children, (child) =>
+					child?.props?.className === '__permalink-h' ? (
+						<MdxHeadingPermalink {...child.props} />
+					) : (
+						child
+					)
+				)}
```

Comment:
For others: Pam and I paired on this briefly, and came to the conclusion that this was a reasonable initial implementation. For a few reasons:

* We control the generated element, via our current remark plugin, that adds the anchor with the `__permalink-h` class, so it should be reliable to detect it in this way
* We know we eventually want the logic to live in the heading component for inserting the permalink anchor, so it felt best to put it here now as opposed to, say, our `MdxA` component.
* Once we do update the remark plugin, this would presumably break, and we can make the necessary adjustments in the implementation.